path.hpp

```diff
@@ -0,0 +1,490 @@
+        return path_.is_relative();
+        }
+
+  private:
+    std::filesystem::path path_;
+
+    friend bool operator==(const path& lhs, const path& rhs) noexcept;
+    friend bool operator!=(const path& lhs, const path& rhs) noexcept;
+    friend bool operator<(const path& lhs, const path& rhs) noexcept;
+    friend bool operator<=(const path& lhs, const path& rhs) noexcept;
+    friend bool operator>(const path& lhs, const path& rhs) noexcept;
+    friend bool operator>=(const path& lhs, const path& rhs) noexcept;
+    friend path operator/(const path& lhs, const path& rhs);
+};
+
+/// Non-member operators.
+
+inline bool operator==(const path& lhs, const path& rhs) noexcept
+{
+    return lhs.path_ == rhs.path_;
+}
+
+inline bool operator!=(const path& lhs, const path& rhs) noexcept
+{
+    return lhs.path_ != rhs.path_;
+}
+
+inline bool operator<(const path& lhs, const path& rhs) noexcept
+{
+    return lhs.path_ < rhs.path_;
+}
+
+inline bool operator<=(const path& lhs, const path& rhs) noexcept
+{
+    return lhs.path_ <= rhs.path_;
+}
+
+inline bool operator>(const path& lhs, const path& rhs) noexcept
+{
+    return lhs.path_ > rhs.path_;
+}
+
+inline bool operator>=(const path& lhs, const path& rhs) noexcept
+{
+    return lhs.path_ >= rhs.path_;
+}
+
+inline path operator/(const path& lhs, const path& rhs)
+{
+    return lhs.path_ / rhs.path_;
+}
+
+template<class CharT, class Traits>
+inline std::basic_ostream<CharT, Traits>&
+operator<<(std::basic_ostream<CharT, Traits>& os, const path& p)
+{
+    return os << p.string();
+}
+
+/// Implement own functions, which return the path.
+
+inline path absolute(const path& p)
+{
+    return std::filesystem::absolute(p);
+}
+
+inline path absolute(const path& p, std::error_code& ec)
+{
+    return std::filesystem::absolute(p, ec);
+}
+
+inline path current_path()
+{
+    return std::filesystem::current_path();
+}
+
+inline path current_path(std::error_code& ec)
+{
+    return std::filesystem::current_path(ec);
+}
+
+/// File streams.
+///
+/// Motivation: our own file streams can be constructed from our own path
+/// class. fs::path has `operator std::filesystem::path()`, but the following
+/// code fails:
+///
+/// fs::path p{"file.ext"};
+/// fs::ifstream s{p};
+///
+/// So implement our own classes for the exact match of constructor parameters.
+///
+/// Note that fs::path can be constructed from a string using fs::u8path()
+/// which leads to differences with the standard file streams constructing:
+///
+/// std::string filename{"unicode_filename.ext"};
+/// fs::ifstream fs1 {filename}; // The filename treated as UTF8 encoded.
+/// std::ifstream fs2{filename}; // Used the current locale.
+
+class LMI_SO ifstream final
+    :public std::ifstream
+{
+  public:
+    ifstream() = default;
+
+    explicit ifstream
+        (path const& p
+        ,std::ios_base::openmode mode = std::ios_base::in
+        )
+        :std::ifstream(static_cast<std::filesystem::path const&>(p), mode)
```

Comment:
I really don't think we should/can do this. Let's provide an explicit accessor instead (or maybe use private inheritance and make this ctor friend of our `path`?).